[
  {
    "captures": {
      "entry": {
        "arguments": {},
        "locals": {},
        "staticFields": {}
      },
      "return": {
        "arguments": {
          "depth": {
            "type": "java.lang.Integer",
            "value": "3"
          },
          "this": {
            "type": "com.datadoghq.system_tests.springboot.ExceptionReplayController",
            "fields": {}
          }
        },
        "locals": {},
        "staticFields": {},
        "throwable": {
          "message": "500 INTERNAL_SERVER_ERROR \"recursion exception depth 3\"",
          "stacktrace": [
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 25
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 23
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 23
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 23
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursion",
              "lineNumber": 18
            },
            {
              "function": "datadog.trace.instrumentation.springweb.HandlerMappingResourceNameFilter.doFilterInternal",
              "lineNumber": 55
            },
            {
              "<runtime>": "<scrubbed>"
            }
          ],
          "type": "org.springframework.web.server.ResponseStatusException"
        }
      }
    },
    "exceptionCaptureId": "bc3df722-9034-44bc-9c10-517863820df2",
    "exceptionHash": "291b2af9246f187556bcc3c598253df517afff6ec78b98c79f42b55fa103c",
    "frameIndex": "4",
    "id": "<scrubbed>",
    "language": "java",
    "probe": {
      "id": "<scrubbed>",
      "version": 0,
      "location": {
        "lines": [],
        "method": "exceptionReplayRecursion",
        "type": "com.datadoghq.system_tests.springboot.ExceptionReplayController"
      }
    },
    "stack": [
      {
        "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursion",
        "lineNumber": 18
      },
      {
        "function": "datadog.trace.instrumentation.springweb.HandlerMappingResourceNameFilter.doFilterInternal",
        "lineNumber": 55
      },
      {
        "<runtime>": "<scrubbed>"
      }
    ],
    "timestamp": "<scrubbed>"
  },
  {
    "captures": {
      "entry": {
        "arguments": {},
        "locals": {},
        "staticFields": {}
      },
      "return": {
        "arguments": {
          "currentDepth": {
            "type": "java.lang.Integer",
            "value": "0"
          },
          "this": {
            "type": "com.datadoghq.system_tests.springboot.ExceptionReplayController",
            "fields": {}
          },
          "originalDepth": {
            "type": "java.lang.Integer",
            "value": "3"
          }
        },
        "locals": {},
        "staticFields": {},
        "throwable": {
          "message": "500 INTERNAL_SERVER_ERROR \"recursion exception depth 3\"",
          "stacktrace": [
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 25
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 23
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 23
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 23
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursion",
              "lineNumber": 18
            },
            {
              "function": "datadog.trace.instrumentation.springweb.HandlerMappingResourceNameFilter.doFilterInternal",
              "lineNumber": 55
            },
            {
              "<runtime>": "<scrubbed>"
            }
          ],
          "type": "org.springframework.web.server.ResponseStatusException"
        }
      }
    },
    "exceptionCaptureId": "bc3df722-9034-44bc-9c10-517863820df2",
    "exceptionHash": "291b2af9246f187556bcc3c598253df517afff6ec78b98c79f42b55fa103c",
    "frameIndex": "0",
    "id": "<scrubbed>",
    "language": "java",
    "probe": {
      "id": "<scrubbed>",
      "version": 0,
      "location": {
        "lines": [],
        "method": "exceptionReplayRecursionHelper",
        "type": "com.datadoghq.system_tests.springboot.ExceptionReplayController"
      }
    },
    "stack": [
      {
        "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
        "lineNumber": 25
      },
      {
        "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
        "lineNumber": 23
      },
      {
        "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
        "lineNumber": 23
      },
      {
        "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
        "lineNumber": 23
      },
      {
        "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursion",
        "lineNumber": 18
      },
      {
        "function": "datadog.trace.instrumentation.springweb.HandlerMappingResourceNameFilter.doFilterInternal",
        "lineNumber": 55
      },
      {
        "<runtime>": "<scrubbed>"
      }
    ],
    "timestamp": "<scrubbed>"
  },
  {
    "captures": {
      "entry": {
        "arguments": {},
        "locals": {},
        "staticFields": {}
      },
      "return": {
        "arguments": {
          "currentDepth": {
            "type": "java.lang.Integer",
            "value": "1"
          },
          "this": {
            "type": "com.datadoghq.system_tests.springboot.ExceptionReplayController",
            "fields": {}
          },
          "originalDepth": {
            "type": "java.lang.Integer",
            "value": "3"
          }
        },
        "locals": {},
        "staticFields": {},
        "throwable": {
          "message": "500 INTERNAL_SERVER_ERROR \"recursion exception depth 3\"",
          "stacktrace": [
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 25
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 23
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 23
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 23
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursion",
              "lineNumber": 18
            },
            {
              "function": "datadog.trace.instrumentation.springweb.HandlerMappingResourceNameFilter.doFilterInternal",
              "lineNumber": 55
            },
            {
              "<runtime>": "<scrubbed>"
            }
          ],
          "type": "org.springframework.web.server.ResponseStatusException"
        }
      }
    },
    "exceptionCaptureId": "bc3df722-9034-44bc-9c10-517863820df2",
    "exceptionHash": "291b2af9246f187556bcc3c598253df517afff6ec78b98c79f42b55fa103c",
    "frameIndex": "1",
    "id": "<scrubbed>",
    "language": "java",
    "probe": {
      "id": "<scrubbed>",
      "version": 0,
      "location": {
        "lines": [],
        "method": "exceptionReplayRecursionHelper",
        "type": "com.datadoghq.system_tests.springboot.ExceptionReplayController"
      }
    },
    "stack": [
      {
        "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
        "lineNumber": 25
      },
      {
        "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
        "lineNumber": 23
      },
      {
        "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
        "lineNumber": 23
      },
      {
        "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursion",
        "lineNumber": 18
      },
      {
        "function": "datadog.trace.instrumentation.springweb.HandlerMappingResourceNameFilter.doFilterInternal",
        "lineNumber": 55
      },
      {
        "<runtime>": "<scrubbed>"
      }
    ],
    "timestamp": "<scrubbed>"
  },
  {
    "captures": {
      "entry": {
        "arguments": {},
        "locals": {},
        "staticFields": {}
      },
      "return": {
        "arguments": {
          "currentDepth": {
            "type": "java.lang.Integer",
            "value": "2"
          },
          "this": {
            "type": "com.datadoghq.system_tests.springboot.ExceptionReplayController",
            "fields": {}
          },
          "originalDepth": {
            "type": "java.lang.Integer",
            "value": "3"
          }
        },
        "locals": {},
        "staticFields": {},
        "throwable": {
          "message": "500 INTERNAL_SERVER_ERROR \"recursion exception depth 3\"",
          "stacktrace": [
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 25
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 23
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 23
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 23
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursion",
              "lineNumber": 18
            },
            {
              "function": "datadog.trace.instrumentation.springweb.HandlerMappingResourceNameFilter.doFilterInternal",
              "lineNumber": 55
            },
            {
              "<runtime>": "<scrubbed>"
            }
          ],
          "type": "org.springframework.web.server.ResponseStatusException"
        }
      }
    },
    "exceptionCaptureId": "bc3df722-9034-44bc-9c10-517863820df2",
    "exceptionHash": "291b2af9246f187556bcc3c598253df517afff6ec78b98c79f42b55fa103c",
    "frameIndex": "2",
    "id": "<scrubbed>",
    "language": "java",
    "probe": {
      "id": "<scrubbed>",
      "version": 0,
      "location": {
        "lines": [],
        "method": "exceptionReplayRecursionHelper",
        "type": "com.datadoghq.system_tests.springboot.ExceptionReplayController"
      }
    },
    "stack": [
      {
        "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
        "lineNumber": 25
      },
      {
        "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
        "lineNumber": 23
      },
      {
        "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursion",
        "lineNumber": 18
      },
      {
        "function": "datadog.trace.instrumentation.springweb.HandlerMappingResourceNameFilter.doFilterInternal",
        "lineNumber": 55
      },
      {
        "<runtime>": "<scrubbed>"
      }
    ],
    "timestamp": "<scrubbed>"
  },
  {
    "captures": {
      "entry": {
        "arguments": {},
        "locals": {},
        "staticFields": {}
      },
      "return": {
        "arguments": {
          "currentDepth": {
            "type": "java.lang.Integer",
            "value": "3"
          },
          "this": {
            "type": "com.datadoghq.system_tests.springboot.ExceptionReplayController",
            "fields": {}
          },
          "originalDepth": {
            "type": "java.lang.Integer",
            "value": "3"
          }
        },
        "locals": {},
        "staticFields": {},
        "throwable": {
          "message": "500 INTERNAL_SERVER_ERROR \"recursion exception depth 3\"",
          "stacktrace": [
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 25
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 23
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 23
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
              "lineNumber": 23
            },
            {
              "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursion",
              "lineNumber": 18
            },
            {
              "function": "datadog.trace.instrumentation.springweb.HandlerMappingResourceNameFilter.doFilterInternal",
              "lineNumber": 55
            },
            {
              "<runtime>": "<scrubbed>"
            }
          ],
          "type": "org.springframework.web.server.ResponseStatusException"
        }
      }
    },
    "exceptionCaptureId": "bc3df722-9034-44bc-9c10-517863820df2",
    "exceptionHash": "291b2af9246f187556bcc3c598253df517afff6ec78b98c79f42b55fa103c",
    "frameIndex": "3",
    "id": "<scrubbed>",
    "language": "java",
    "probe": {
      "id": "<scrubbed>",
      "version": 0,
      "location": {
        "lines": [],
        "method": "exceptionReplayRecursionHelper",
        "type": "com.datadoghq.system_tests.springboot.ExceptionReplayController"
      }
    },
    "stack": [
      {
        "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursionHelper",
        "lineNumber": 25
      },
      {
        "function": "com.datadoghq.system_tests.springboot.ExceptionReplayController.exceptionReplayRecursion",
        "lineNumber": 18
      },
      {
        "function": "datadog.trace.instrumentation.springweb.HandlerMappingResourceNameFilter.doFilterInternal",
        "lineNumber": 55
      },
      {
        "<runtime>": "<scrubbed>"
      }
    ],
    "timestamp": "<scrubbed>"
  }
]